test/unit/buffer_spec.lua

```diff
@@ -212,92 +213,164 @@ describe('buffer functions', function()
+                                       false,
+                                       arg.fillchar,
+                                       arg.maxwidth,
+                                       NULL,
+                                       NULL)
+      end
     end)
 
+    local STL_MAX_ITEM = 80
+
+    -- TODO: Incorporate these into the new format, if possible.
     it('should print no file name', function()
-      local width = build_stl_str_hl("%f")
+      local width = build_stl_str_hl{ pat="%f", maxwidth=80 }
 
       eq(9, width)
       eq("[No Name]", helpers.ffi.string(output_buffer, width))
-
     end)
 
     it('should print the relative file name', function()
       buffer.setfname(globals.curbuf, to_cstr("Makefile"), NULL, 1)
-      local width = build_stl_str_hl("%f")
+      local width = build_stl_str_hl{ pat="%f", maxwidth=80 }
 
       eq(8, width)
       eq("Makefile", helpers.ffi.string(output_buffer, width))
-
     end)
 
     it('should print the full file name', function()
       buffer.setfname(globals.curbuf, to_cstr("Makefile"), NULL, 1)
 
-      local width = build_stl_str_hl("%F")
+      local width = build_stl_str_hl{ pat="%F", maxwidth=80 }
 
       assert.is_true(8 < width)
       neq(NULL, string.find(helpers.ffi.string(output_buffer, width), "Makefile"))
 
     end)
 
-    it('should print the tail file name', function()
-      buffer.setfname(globals.curbuf, to_cstr("src/nvim/buffer.c"), NULL, 1)
-
-      local width = build_stl_str_hl("%t")
-
-      eq(8, width)
-      eq("buffer.c", helpers.ffi.string(output_buffer, width))
+    it('should correctly input different fillchars', function()
+      local actual_width = build_stl_str_hl{ pat='abcde%=', maxwidth=10, fillchar=0x21 }
 
+      eq('abcde!!!!!', get_str(output_buffer, 10))
     end)
 
-    it('should print the buffer number', function()
-      buffer.setfname(globals.curbuf, to_cstr("src/nvim/buffer.c"), NULL, 1)
+    local statusline_test = function(name, input_width, expected_width, stl, expected_stl)
+      it(name, function()
+        buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
 
-      local width = build_stl_str_hl("%n")
+        local actual_width = build_stl_str_hl{ pat=stl, maxwidth=input_width }
 
-      eq(1, width)
-      eq("1", helpers.ffi.string(output_buffer, width))
-    end)
-
-    it('should print the current line number in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%l")
-
-      eq(1, width)
-      eq("0", helpers.ffi.string(output_buffer, width))
-
-    end)
-
-    it('should print the number of lines in the buffer', function()
-      buffer.setfname(globals.curbuf, to_cstr("test/unit/buffer_spec.lua"), NULL, 1)
-
-      local width = build_stl_str_hl("%L")
+        eq(actual_width, expected_width)
+        eq(expected_stl, get_str(output_buffer, expected_width))
```

Comment:
This is not correct. AFAIK `output_buffer` is NUL-terminated, so `expected_width` (which is width *in cells*, not *byte length*) is not needed here *at all*.